Source/ASNetworkImageNode.mm

```diff
@@ -696,50 +696,61 @@ - (void)_lazilyLoadImageIfNecessary
     } else {
       __weak __typeof__(self) weakSelf = self;
       auto finished = ^(id <ASImageContainerProtocol>imageContainer, NSError *error, id downloadIdentifier) {
-       
-        __typeof__(self) strongSelf = weakSelf;
-        if (strongSelf == nil) {
-          return;
-        }
-
-        as_log_verbose(ASImageLoadingLog(), "Downloaded image for %@ img: %@ urls: %@", self, [imageContainer asdk_image], URLs);
-        
-        // Grab the lock for the rest of the block
-        ASDN::MutexLocker l(strongSelf->__instanceLock__);
-        
-        //Getting a result back for a different download identifier, download must not have been successfully canceled
-        if (ASObjectIsEqual(strongSelf->_downloadIdentifier, downloadIdentifier) == NO && downloadIdentifier != nil) {
-          return;
-        }
+        ASPerformBlockOnBackgroundThread(^{
```

Comment:
Just to confirm, the only change here is that the (previous) `finished` block is now executed on a background thread? (it's a bit hard to read)

Reply:
Mostly. Everything before was happening on main. Now it's being dispatched to the background and once it's done the work, the delegate callbacks are dispatched to main.